tools/automation/vendoring/patches/importlib-metadata.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/src/pip/_vendor/importlib_metadata/__init__.py b/src/pip/_vendor/importlib_metadata/__init__.py
+index 089fca97..50fcea99 100644
+--- a/src/pip/_vendor/importlib_metadata/__init__.py
++++ b/src/pip/_vendor/importlib_metadata/__init__.py
+@@ -6,7 +6,7 @@ import re
+ import abc
+ import csv
+ import sys
+-import zipp
++from pip._vendor import zipp
+ import operator
+ import functools
+ import itertools
+@@ -586,6 +586,3 @@ def requires(distribution_name):
+     packaging.requirement.Requirement.
+     """
+     return distribution(distribution_name).requires
+-
+-
+-__version__ = version(__name__)
```

Comment:
The `version` command looks for the `importlib-metadata` distribution itself in `sys.path`, and breaks with vendoring since there’s no metadata anywhere to be found. I figure it’s easier to remove the global call (which fails in import time) since we have no interest in reading `__version__` anyway.

Reply:
A comment describing this somewhere on the patch, would be useful. Instead of removing the lines, consider commenting them out and noting why, similar to how we do it in `requests.patch` for the somplejson upload. :)

Reply:
Yup, as reminded in #7784 patches should be dcumented/motivated here: https://github.com/pypa/pip/blob/master/src/pip/_vendor/README.rst#modifications

Reply:
An concerning the version value, it might actually be useful if we wanted to include the vendored libraries' version in `pip debug` output to help diagnose _tweaked_ installation. But that's not currently the case indeed.